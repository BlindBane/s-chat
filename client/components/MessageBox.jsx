import React from 'react';
import Message from './Message';

function MessageBox(props) {
  const messages = props.messages.map(msg => (
    <Message
      key={msg.userId}
      username={msg.username}
      messageText={msg.messageText}
      channel={msg.channel}
    />
  ));
  return (
    <div>
      {messages}
    </div>
  );
}

export default MessageBox;
