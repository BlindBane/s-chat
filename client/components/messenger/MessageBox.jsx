import React from 'react'
import Message from './Message'
import MessageInput from './MessageInput'

function MessageBox (props) {
  const messages = props.messages.filter(
      (msg) =>
        `${msg.username} ${msg.channel} ${msg.messageText}`.toUpperCase()
        .indexOf(props.searchTerm.toUpperCase()) >= 0
    )
    .map(msg => (
      <Message
        key={msg.userId}
        username={msg.username}
        messageText={msg.messageText}
        channel={msg.channel}
      />
    ))
  return (
    <div>
      {messages}
      <MessageInput handleNewMsg={props.handleNewMsg} />
    </div>
  )
}

export default MessageBox
