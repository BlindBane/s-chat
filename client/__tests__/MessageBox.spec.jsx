import React from 'react'
import { shallow } from 'enzyme'
import MessageBox from './../components/messenger/MessageBox'

import msgs from './../../data.json'

test('MessageBox renders correctly', () => {
  const searchTerm = ''
  const component = shallow(<MessageBox messages={msgs.messages} searchTerm={searchTerm} />)
  expect(component).toMatchSnapshot()
})
