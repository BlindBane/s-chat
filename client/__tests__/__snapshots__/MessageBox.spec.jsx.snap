// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox renders correctly 1`] = `
<div>
  <Message
    channel="sanFrancisco"
    messageText="Hi, I am bill."
    username="Bill"
  />
  <Message
    channel="general"
    messageText="Hi, I am Bob."
    username="Bob"
  />
  <Message
    channel="ASU"
    messageText="Hi, I am Bert."
    username="Bert"
  />
  <MessageInput />
</div>
`;
